src/contexts/locales.jsx

```diff
@@ -0,0 +1,76 @@
+/* eslint-env browser */
+import React, { createContext, useEffect } from 'react'
+import useLocalStorage from 'react-use/lib/useLocalStorage'
+import PropTypes from 'prop-types'
+
+const getFromPath = (data, path) => path.split('.').reduce(
+  (curr, sub) => curr && curr[sub],
+  data,
+)
+
+const Context = createContext()
+
+const LocalesProvider = ({ children }) => {
+  const [lang, setLang] = useLocalStorage('lang', undefined)
+  const [locales, setLocales] = useLocalStorage('locales', undefined)
+
+  useEffect(() => {
+    // load lang
+    let currentLang = lang || 'en'
```

Comment:
if lang is undefined, we will run into the if statement below, and finally end up with 'en' if nothing is found.
So I think the default value is also not relevant here